consensus/XDPoS/engines/engine_v2/engine.go

```diff
@@ -121,48 +124,72 @@ func (x *XDPoS_v2) SignHash(header *types.Header) (hash common.Hash) {
 	return sigHash(header)
 }
 
-func (x *XDPoS_v2) Initial(chain consensus.ChainReader, masternodes []common.Address) error {
+func (x *XDPoS_v2) Initial(chain consensus.ChainReader, header *types.Header) error {
 	log.Info("[Initial] initial v2 related parameters")
 
 	if x.highestQuorumCert.ProposedBlockInfo.Hash != (common.Hash{}) { // already initialized
 		log.Error("[Initial] Already initialized", "x.highestQuorumCert.ProposedBlockInfo.Hash", x.highestQuorumCert.ProposedBlockInfo.Hash)
 		return nil
 	}
 
-	x.lock.Lock()
-	defer x.lock.Unlock()
-	// Check header if it is the first consensus v2 block, if so, assign initial values to current round and highestQC
-
-	log.Info("[Initial] highest QC for consensus v2 first block")
-	// Generate new parent blockInfo and put it into QC
-	// TODO: XIN-147 to initilise V2 engine in a more dynamic way
-	firstV2BlockHeader := chain.GetHeaderByNumber(x.config.V2.SwitchBlock.Uint64())
-	blockInfo := &utils.BlockInfo{
-		Hash:   firstV2BlockHeader.Hash(),
-		Round:  utils.Round(0),
-		Number: firstV2BlockHeader.Number,
-	}
-	quorumCert := &utils.QuorumCert{
-		ProposedBlockInfo: blockInfo,
-		Signatures:        nil,
+	if header.Number.Int64() == x.config.V2.SwitchBlock.Int64() {
+		log.Info("[Initial] highest QC for consensus v2 first block")
+		firstV2BlockHeader := chain.GetHeaderByNumber(header.Number.Uint64())
+		blockInfo := &utils.BlockInfo{
+			Hash:   firstV2BlockHeader.Hash(),
+			Round:  utils.Round(0),
+			Number: firstV2BlockHeader.Number,
+		}
+		quorumCert := &utils.QuorumCert{
+			ProposedBlockInfo: blockInfo,
+			Signatures:        nil,
+		}
+		x.currentRound = 1
```

Comment:
how about just call `processQC(quorumCert)`?

Reply:
cannot use processQC, because round number is same as default. And maybe change to Nil on highQC, or set negative value Round Uint64 -> Int64. Either of way doesn't look good